### DISCUSSION TITLE
CORS Origin and port configuration for tauri desktop application

### DISCUSSION THREAD

**Discussion by VassilisM** [tauri-apps/tauri#6898] [Q&A]
Hello, I've been trying to find out what is the origin sent by a tauri desktop application (Tauri+Vue+TS) when performing rest requests, and if I can configure it. The application does nothing rust-wise; just avails the VueTS front-end as a desktop application.

The responsible (rust) back-end server (independent from this tauri application) shows incoming requests from "-" (see below) which are being denied, of course. ".allowed_origin("-")" does not work and ".allow_any_origin()" is **not** an option.

Is there a way to configure the origin of the application in order to add it properly in allowed origins of the cors setup in the server side?

On a second note, regarding the port being used while "npm run tauri:serve" for develop/debugging, is there a way to configure the "http://localhost:8080/" (sometimes http://localhost:8081/)?

Thank you!

[2023-04-28T09:05:40Z INFO actix_web::middleware::logger] 127.0.0.1 "OPTIONS /api/v1/test HTTP/1.1" 400 42 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36 Edg/112.0.1722.58" 0.001110

**Comment by VassilisM** [comment]
Answers from here: https://github.com/tauri-apps/tauri-docs/issues/26
only suggest bypassing CORS. (If I guess correctly: Tauri as client.)

Is there really no way of setting the origin as it is not supposed to be a "browser"?

If no origin configuration is possible I guess there is no other option than creating rust methods to perform the rest calls and return the responses to the front-end.

UPDATE: Using a wrapper for the Tauri Http, which uses rust methods, could also work.
-> It works, but it is a pain resolving the (tauri)Response<T> as normal Response.

**Comment by FabianLars** [comment]
Are you on Linux? If so, can you try the `linux-headers` feature flag? afaik the origin should never be `-`, i can't even remember seeing that tbh, only completely empty ones but maybe actix fills that in for empty origins?

Other than that it is not possible to configure the origin for the webview. (with the exception of windows: https://github.com/tauri-apps/tauri/discussions/4912#discussioncomment-3406282)
